File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        a, b = 0, 0
        n = len(colors)
        for i in range(2, n):
            if colors[i] == colors[i-1] and colors[i] == colors[i-2]:
                if colors[i] == 'A':
                    a += 1
                else:
                    b += 1
        return a > b
            

File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.cpp
class Solution {
public:
    bool winnerOfGame(string colors) {
        int n = colors.size();
        int a = 0, b = 0;
        for (int i = 2; i < n; i++) {
            if (colors[i] == colors[i-1] && colors[i] == colors[i-2]) {
                if (colors[i] == 'A') {
                    a++;
                }
                else {
                    b++;
                }
            }
        }
        return a > b;
    }
};